pkg/receive/handler.go

```diff
@@ -229,6 +237,18 @@ func (h *Handler) TransformAndValidateWriteRequest(next http.Handler, labels ...
 			return
 		}
 
+		if len(wreq.GetTimeseries()) == 0 && len(wreq.GetMetadata()) > 0 {
+			// thanos does not yet support metadata requests, so we silently accept and discard them here.
+			// we may want to add a flag to enable this in the future when thanos can consume metadata requests via remote write.
+			h.metadataRequestsTotal.Inc()
+
+			w.WriteHeader(http.StatusOK)
```

Comment:
Is this how we manage the status on the write request? What if it fails after telemeter for any other reason - would this misreport a 200?

Reply:
this is a case of exiting early if we dont support the request. we want to accept it and discard it in the case of metadata only. protected by the if statement above it.